app/core/Analytics/MetricsEventBuilder.ts

```diff
@@ -77,15 +77,45 @@ class MetricsEventBuilder {
     return new MetricsEventBuilder(event);
   }
 
+  #getFilteredLegacyProperties = (properties: JsonMap) => {
+    const [filteredProps, anonymousProps] = Object.entries(properties).reduce(
```

Comment:
```suggestion
    const filteredProps: JsonMap = {};
const anonymousProps: JsonMap = {};

for (const [key, value] of Object.entries(properties)) {
  if (
    value !== null && 
    typeof value === 'object' && 
    'anonymous' in value &&
    value.anonymous === true
  ) {
    anonymousProps[key] = value.value;
  } else {
    filteredProps[key] = value;
  }
}

return [filteredProps, anonymousProps];
```

Save on performance by removing reduce